File: src/BreakfastProvider.Api/Models/Requests/IngredientUsageRequest.cs
namespace BreakfastProvider.Api.Models.Requests;

public record IngredientUsageRequest
{
    public string? IngredientName { get; init; }
    public decimal QuantityUsed { get; init; }
    public string? Unit { get; init; }
    public string? RecipeName { get; init; }
}

File: java/breakfast-provider-app/src/main/java/io/lemonlion/breakfast/model/request/IngredientUsageRequest.java
package io.lemonlion.breakfast.model.request;

import java.math.BigDecimal;

/** Twin of C# {@code IngredientUsageRequest}. */
public record IngredientUsageRequest(String ingredientName, BigDecimal quantityUsed, String unit, String recipeName) {

    public IngredientUsageRequest {
        if (quantityUsed == null) {
            quantityUsed = BigDecimal.ZERO;
        }
    }
}
